kratos/tests/cpp_tests/utilities/test_split_tetrahedra.cpp

```diff
@@ -0,0 +1,233 @@
+        const unsigned int i_0 = r_i_face[0].Id();
+        const unsigned int i_1 = r_i_face[1].Id();
+        const unsigned int i_2 = r_i_face[2].Id();
+
+        // Check the current face against all faces
+        for (unsigned int j_face = 0; j_face < rSubTetrasFaces.size(); ++j_face) {
+            const Geometry<Node<3>> &r_j_face = rSubTetrasFaces[j_face];
+            const unsigned int j_0 = r_j_face[0].Id();
+            const unsigned int j_1 = r_j_face[1].Id();
+            const unsigned int j_2 = r_j_face[2].Id();
+
+            if (i_0 == j_0 || i_0 == j_1 || i_0 == j_2) {
+                if (i_1 == j_0 || i_1 == j_1 || i_1 == j_2) {
+                    if (i_2 == j_0 || i_2 == j_1 || i_2 == j_2) {
+                        i_face_times++;
+                    }
+                }
+            }
+        }
+
+        // If the face it's external, there must be a unique occurrence
+        if (i_face_times == 1) {
+            rSubTetrasExtFaces.push_back(r_i_face);
+        }
+    }
+}
+
+void PrintEdgesAndExternalFaces(
+    const unsigned int Permutation,
+    const std::vector<int> &rEdgeIntVect,
+    const std::vector<Geometry<Node<3>>> rSubTetrasExtFaces)
+{
+    std::cout << "(("
+        << rEdgeIntVect[0] << ","
+        << rEdgeIntVect[1] << ","
+        << rEdgeIntVect[2] << ","
+        << rEdgeIntVect[3] << ","
+        << rEdgeIntVect[4] << ","
+        << rEdgeIntVect[5] << ") , [";
+
+    for (unsigned int i_ext_face = 0; i_ext_face < rSubTetrasExtFaces.size(); i_ext_face++) {
+        const Geometry<Node<3>> &r_i_ext_face = rSubTetrasExtFaces[i_ext_face];
+        const unsigned int i_ext_0 = r_i_ext_face[0].Id();
+        const unsigned int i_ext_1 = r_i_ext_face[1].Id();
+        const unsigned int i_ext_2 = r_i_ext_face[2].Id();
+        std::cout << "(" << i_ext_0 << "," << i_ext_1 << "," << i_ext_2 << ")";
+        if (i_ext_face != (rSubTetrasExtFaces.size() - 1)) {
+            std::cout << ",";
+        }
+    }
+
+    if (Permutation != 729) {
+        std::cout << "])," << std::endl;
+    } else {
+        std::cout << "])"  << std::endl;
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TetrahedraSplitModes, KratosCoreFastSuite)
+{
+    std::vector<int> aux_ids(11);
+    std::vector<int> edge_ids(6);
+
+    aux_ids[0] = 2; aux_ids[1] = 1; aux_ids[2] = 4; aux_ids[3] = 3; // Vertices
+    aux_ids[4] = -1; // Edge 01
+    aux_ids[5] =  1; // Edge 02 (split) 
+    aux_ids[6] = -1; // Edge 03
+    aux_ids[7] =  1; // Edge 12 (split)
+    aux_ids[8] = -1; // Edge 13
+    aux_ids[9] =  1; // Edge 23 (split)
+    aux_ids[10] = -1; // Internal (Steiner) node not required
+
+    TetrahedraSplit::TetrahedraSplitMode(aux_ids.data(), edge_ids.data());
+
+    KRATOS_CHECK_EQUAL(edge_ids[0], 0);
+    KRATOS_CHECK_EQUAL(edge_ids[1], 5);
+    KRATOS_CHECK_EQUAL(edge_ids[2], 3);
+    KRATOS_CHECK_EQUAL(edge_ids[3], 7);
+    KRATOS_CHECK_EQUAL(edge_ids[4], 3);
+    KRATOS_CHECK_EQUAL(edge_ids[5], 9);
```

Comment:
minor comment, I think all those macros increase the compilation time (pls correct me if I am wrong @roigcarlo )

alternative solutions.
~~~c
const std::vector<int> expected_results{0,5,3,7,3,9 }; // maybe constexpr, not sure
for (std::size_t i=0; i<6; ++i) {
    KRATOS_CHECK_EQUAL(edge_ids[i], expected_results[i]);
}
~~~

this way you only have to compile the macro once

Reply:
Good to know! Thanks @philbucher